app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,73 @@
+class OrderItemsController < ApplicationController
+  before_action :set_order, only: [:index, :destroy]
+
+  def index
+    @order_items = @order.order_items || []
+  end
+  # def index
+  #   @order_items = current_user.order_items # or the correct association
+  #   @OrderItem = @order_items.first # Example logic, adjust as needed
+  # end
+
+  def create
+    @order = current_user.orders.find_or_create_by(status: :cart)
+    product = Product.find(params[:product_id])
+    order_item = @order.order_items.find_or_initialize_by(product: product)
+    if order_item.new_record?
+      order_item.quantity = 1
+    else
+      order_item.quantity += 1
+    end
```

Comment:
You can refactor your condition like that `order_item.quantity = order_item.new_record? ?  1 : order_item.quantity + 1`